docs/content/dev/authentication.md

```diff
@@ -0,0 +1,45 @@
+# User Profiles and Authentication
+Each user in HedgeDoc 2 has a profile
+which contains the following information:
+
+- username (`janedoe`)
+- display (`Jane Doe`)
+- email address, optional (`janedoe@example.com`)
+- profile picture, optional
+- the date the user was created
+- the date the profile was last updated 
+
+HedgeDoc 2 supports multiple authentication methods per user.  
+These are called *identities* and each identity is backed by an
+auth provider (like OAuth, SAML, LDAP or internal auth).
+
+One of a users identities may be marked as *sync source*.  
+This identity is used to automatically update profile attributes like the
+display name or profile picture. If a sync source exists, the
+user can not manually edit their profile information.
+If an external provider was used to create the account,
+it is automatically used as sync source. 
+
+The administrator may globally set an auth provider as sync source,
+e.g. to enforce that all profile information comes from the corporate
+LDAP and is the same across multiple applications.  
+If a global sync source exists, new accounts can only be created using
```

Comment:
This rules out the ability to use multiple LDAP backends in parallel. So might be worth noting.

Reply:
This is an interesting point.  
We could allow multiple global sync sources which can be used for sign up and then pin the newly created user to the provider that was used to create the account.

This will make the handling more complicated, but I fear our planned support for multiple auth providers of the same type will be somewhat useless unless we allow multiple sign up-sources.